### PR TITLE
Adding errors for receive and transmit read/write async

### PR DESCRIPTION
**Pull request by MabezDev**
### Submission Checklist 📝
- [x] I have updated existing examples or added new ones (if applicable).
- [x] My changes were added to the [`CHANGELOG.md`](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/CHANGELOG.md) in the **_proper_** section.
#### Extra:
- [x] I have read the [CONTRIBUTING.md guide](https://github.com/esp-rs/esp-hal/blob/main/CONTRIBUTING.md) and followed its instructions.

#### Description

Completion of https://github.com/esp-rs/esp-hal/pull/1168, read the discussion there.

Thanks again @konsulten!

### CODE REVIEW

**Review by JurajSadel — APPROVED** [review]
LGTM, thanks!

File: esp-hal/src/uart.rs [hunk 1]
```diff
@@ -544,6 +550,13 @@ where
             }
         }
 
+        // Setting err_wr_mask stops uart from storing data when data is wrong according
+        // to reference manual
+        T::register_block()
+            .conf0()
+            .modify(|_, w| w.err_wr_mask().set_bit());
+        // TODO reset? https://github.com/espressif/esp-idf/blob/4d90eedb6ef32ff949c6f298a3d845f9f07bac92/components/hal/esp32s3/include/hal/uart_ll.h#L112
```
Comment (bjoernQ) [inline esp-hal/src/uart.rs:558]: IIRC that is done in `uart_peripheral_reset` which is called by `init` earlier. There were problems with UART0 if console logging is enabled

Comment (MabezDev) [reply esp-hal/src/uart.rs]: Good spot, thank you! I've removed the comment.

**Review by bjoernQ — APPROVED** [review]
LGTM

**Pull request merged**